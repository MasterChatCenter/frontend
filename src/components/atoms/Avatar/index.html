<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="./style.css">
    <title>Avatar</title>
</head>
<body>
    <div class="personal__image">
        <label class="label">
            <input type="file">
            <figure class="personal__figure">
                <img class="personal__avatar" src="https://cdn.pixabay.com/photo/2018/01/15/07/51/woman-3083383_960_720.jpg" alt="profile picture">
                <figcaption class="personal__figcaption">
                    <img  class="personal__figcaption--camera" src="./Assets/camera.svg" alt="personal figcaption">
                </figcaption>
            </figure>
        </label>
    </div>
</body>
</html>