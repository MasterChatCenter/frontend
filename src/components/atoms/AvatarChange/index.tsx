import React, { FC } from 'react';
import { FaCamera } from 'react-icons/fa';
import { Label, Figure, Figcation } from './styles';

type props = {
  url: string;
  alt: string;
  handleChange?: any;
};

const AvatarUser: FC<props> = ({ url, alt, handleChange }) => {
  return (
    <Label>
      <input type="file" onChange={handleChange} />
      <Figure>
        <img src={url} alt={alt} />
        <Figcation>
          <FaCamera />
        </Figcation>
      </Figure>
    </Label>
  );
};
export default AvatarUser;
