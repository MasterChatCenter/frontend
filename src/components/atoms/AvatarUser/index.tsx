import React from 'react';
import { AvatarUserStyle } from './styles';

const AvatarUser: React.FC<{ name: string, avatarurl: string }> = ({
  name,
  avatarurl,
}) => {
  return (
    <AvatarUserStyle src={avatarurl} alt={name} />
  )
}
export default AvatarUser