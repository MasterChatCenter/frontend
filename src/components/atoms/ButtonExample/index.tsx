import React, { FC } from 'react';
import { CSSButton } from './styles';

type props = {
  children: string,
};

const ButtonExample: FC<props> = ({ children }) => (
  <CSSButton>
    {children}
  </CSSButton>
);

export default ButtonExample;