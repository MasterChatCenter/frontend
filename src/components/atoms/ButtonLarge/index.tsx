import React, { FC } from 'react';

import { CSSButton } from './styles';

type props = {
  children: string,
  type?: 'button' | 'submit' | 'reset',
};

const ButtonLarge: FC<props> = ({ children, type }) => (
  <CSSButton type={type}>
    {children}
  </CSSButton>
);

ButtonLarge.defaultProps = {
  type: 'button',
};

export default ButtonLarge;
