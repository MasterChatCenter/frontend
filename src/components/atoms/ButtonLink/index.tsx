import React, { FC } from 'react';

import { CSSButton } from './styles';

type props = {
  children: string;
  type?: 'button' | 'submit' | 'reset';
};

const ButtonLink: FC<props> = ({ children, type }) => (
  <CSSButton type={type}>{children}</CSSButton>
);

ButtonLink.defaultProps = {
  type: 'button',
};

export default ButtonLink;
