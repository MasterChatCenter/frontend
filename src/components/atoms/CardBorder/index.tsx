import React, { FC, ReactNode } from 'react';

import { CSSCardBorder } from './styles';

type props = {
  children: ReactNode;
};

const CardBorder: FC<props> = ({ children }) => (
  <CSSCardBorder>{children}</CSSCardBorder>
);

export default CardBorder;
