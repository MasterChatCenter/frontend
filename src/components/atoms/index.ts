export { default as AvatarUser } from './AvatarUser';
export { default as AvatarChange } from './AvatarChange';
export { default as CardBorder } from './CardBorder';
export { default as CardShadow } from './CardShadow';
export { default as InputSearch } from './InputSearch';
export { default as InputEdit } from './InputEdit';
export { default as InputText } from './InputText';
export { default as InputSelect } from './InputSelect';
export { default as ButtonLarge } from './ButtonLarge';
export { default as ButtonLink } from './ButtonLink';
export { default as Loading } from './Loading';
export { default as NavItem } from './NavItem';
