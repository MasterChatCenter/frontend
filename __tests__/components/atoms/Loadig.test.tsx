import React from 'react';
import { Loading } from '../../../src/components/atoms';
import { render } from '../../utils';

describe('atoms / Loading', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Loading />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
