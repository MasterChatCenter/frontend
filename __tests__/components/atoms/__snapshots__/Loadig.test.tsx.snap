// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms / Loading matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="styles__Container-sc-1x8w729-0 kYCmay"
  >
    <div
      class="styles__Ring-sc-1x8w729-1 dlEWCs"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
